src/routes/api/registry/index.ts

```diff
@@ -634,3 +634,85 @@ export const unregisterAgentDelete = payAuthenticatedEndpointFactory.build({
     };
   },
 });
+
+export const deleteAgentRegistrationSchemaInput = z.object({
+  id: z
+    .string()
+    .cuid()
+    .describe(
+      'The database ID of the agent registration record to be deleted.',
+    ),
+});
+
+export const deleteAgentRegistrationSchemaOutput = z.object({
+  success: z.boolean(),
+  message: z.string(),
+});
+
+export const deleteAgentRegistration = payAuthenticatedEndpointFactory.build({
```

Comment:
I would change it to admin authenticated endpoint to delete, then you can also remove the network check.